src/java_bytecode/java_bytecode_convert_class.cpp

```diff
@@ -233,9 +233,20 @@ void java_bytecode_convert_classt::convert(const classt &c)
       extract_generic_superclass_reference(c.signature);
     if(superclass_ref.has_value())
     {
-      const java_generic_symbol_typet generic_base(
-        base, superclass_ref.value(), qualified_classname);
-      class_type.add_base(generic_base);
+      try
+      {
+        const java_generic_symbol_typet generic_base(
+          base, superclass_ref.value(), qualified_classname);
+        class_type.add_base(generic_base);
+      }
+      catch(unsupported_java_class_signature_exceptiont)
+      {
+        debug() << "unsupported generic superclass signature "
+                << id2string(*superclass_ref)
+                << " / original signature "  << id2string(*c.signature)
```

Comment:
`c.signature` is the full signature of the derived class (that includes its generic parameter types, and the superclass and interfaces references). I would not print that out, only the extracted superclass reference, as you do in the case of interfaces below.

Reply:
ok, removed debug output of `c.signature`